src/python_testing/TC_CADMIN_1_3_4.py

```diff
@@ -0,0 +1,389 @@
+                     "Verify DUT_CE responds to both write/read with a success"),
+            TestStep(4, "TH creates a controller (TH_CR2) on a new fabric and commissions DUT_CE using that controller. TH_CR2 should commission the device using a different NodeID than TH_CR1.",
+                     "Commissioning is successful"),
+            TestStep(5, "TH_CR1 reads the Fabrics attribute from the Node Operational Credentials cluster using a fabric-filtered read",
+                     "Verify that the RootPublicKey matches the root public key for TH_CR1 and the NodeID matches the node ID used when TH_CR1 commissioned the device."),
+            TestStep(6, "TH_CR2 reads the Fabrics attribute from the Node Operational Credentials cluster using a fabric-filtered read",
+                     "Verify that the RootPublicKey matches the root public key for TH_CR2 and the NodeID matches the node ID used when TH_CR2 commissioned the device."),
+            TestStep(7, "TH_CR2 reads the CurrentFabricIndex attribute from the Operational Credentials cluster and saves as th2_idx, TH_CR1 sends the RemoveFabric command to the DUT with the FabricIndex set to th2_idx",
+                     "TH_CR1 removes TH_CR2 fabric using th2_idx")
+        ]
+
+    @async_test_body
+    async def test_TC_CADMIN_1_4(self):
+        self.step(1)
+
+        # Establishing TH1
+        self.th1 = self.default_controller
+
+        self.step(2)
+        GC_cluster = Clusters.GeneralCommissioning
+        attribute = GC_cluster.Attributes.BasicCommissioningInfo
+        duration = await self.read_single_attribute_check_success(endpoint=0, cluster=GC_cluster, attribute=attribute)
+        self.max_window_duration = duration.maxCumulativeFailsafeSeconds
+
+        self.step("3a")
+        obcCmd = Clusters.AdministratorCommissioning.Commands.OpenBasicCommissioningWindow(180)
+        await self.th1.SendCommand(nodeid=self.dut_node_id, endpoint=0, payload=obcCmd, timedRequestTimeoutMs=6000)
+
+        self.step("3b")
+        services = await self.get_txt_record()
+        if services.txt_record['CM'] != "1":
+            asserts.fail(f"Expected cm record value not found, instead value found was {str(services.txt_record['CM'])}")
+
+        self.step("3c")
+        BI_cluster = Clusters.BasicInformation
+        nl_attribute = BI_cluster.Attributes.NodeLabel
+        await self.write_nl_attr(th=self.th1, attr_val=nl_attribute)
+        await self.read_nl_attr(th=self.th1, attr_val=nl_attribute)
+
+        self.step(4)
+        # Establishing TH2
+        th2_certificate_authority = self.certificate_authority_manager.NewCertificateAuthority()
+        th2_fabric_admin = th2_certificate_authority.NewFabricAdmin(vendorId=0xFFF1, fabricId=self.th1.fabricId + 1)
+        self.th2 = th2_fabric_admin.NewController(nodeId=2)
+        setupPayloadInfo = self.get_setup_payload_info()
```

Comment:
So this is a bit of a dragon that won't be immediately apparent in the CI. get_setup_payload_info won't necessarily have anything populated if the device is already commissioned when this test starts. So you'll have to enforce that the tester supplies the setup_payload in some form even if the device is already commissioned (ie test run with no factory reset). If you could make this error and the explanation of why this is required VERY explicit, it will be helpful at cert. So two suggestions here:
1) move this up to the top, verify that you have something returned and if not assert with a very explicit message that the setup payload is required for this test even if the device is already commissioned because it is testing the basic commissioning window, which uses the original passcode and discriminator
2) add an extra run of this test to the CI that doesn't use factory reset, to be run right after the original run to verify this works with a pre-commissioned device.

Reply:
Hi Cecille,
Thank you for the suggestions and tips on this situation above, I have made the changes to the script to hopefully resolve the above comments suggestions, please let me know if this works to what you were expecting and if you think it will be acceptable.